### ISSUE TITLE
What is the quantitative method we use? Symmetric quantization or asymmetric quantization

### ISSUE THREAD

**Issue by YZW-explorer** [Xilinx/brevitas#299]
What is the quantitative method we use? Symmetric quantization or asymmetric quantization

**Comment by volcacius** [comment]
Hello,
It depends on the quantizer. In general what makes a quantizer asymmetric is whether it's signed or not and whether zero-point is zero or not.

**Comment by YZW-explorer** [comment]
Thank you very much! I'm looking at the source code of brevitas, can you tell me which part of the code is “quant_weight = self.quant_weight()”? Due to the encapsulation of the code, I cannot find it.

**Comment by volcacius** [comment]
you mean this? https://github.com/Xilinx/brevitas/blob/9a5875390f4cf9720c64567d72a83c1023d4f951/src/brevitas/nn/quant_layer.py#L345

**Comment by YZW-explorer** [comment]
Yes, but I cannot find the underlying implementation of this function.

**Comment by volcacius** [comment]
It works this way:
- Here you are defining self.quant_weight() https://github.com/Xilinx/brevitas/blob/9a5875390f4cf9720c64567d72a83c1023d4f951/src/brevitas/nn/mixin/parameter.py#L90
- Here you are defining self.weight_quant(...) 
https://github.com/Xilinx/brevitas/blob/9a5875390f4cf9720c64567d72a83c1023d4f951/src/brevitas/nn/mixin/parameter.py#L64 
https://github.com/Xilinx/brevitas/blob/9a5875390f4cf9720c64567d72a83c1023d4f951/src/brevitas/nn/mixin/base.py#L109

Basically what happens is that self.weight_quant is an instance of `WeightQuantProxyFromInjector` which is a wrapper for a quantizer like `Int8WeightPerTensorFloat`.
 
My suggestion if you want to make sure you understand how it works it's to just go through the whole execution with a step-by-step debugger.

**Comment by YZW-explorer** [comment]
thank you so much!

**Comment by YZW-explorer** [comment]
Sincerely thank you for your explanation！
